v2/chunk.go

```diff
@@ -0,0 +1,35 @@
+package pie
+
+// Chunk returns slices with each length equalling chunkLength, except for the last slice which may contain less.
+// 	The order is maintained in each of the slices and slices themselves.
+//
+// Examples:
+//
+// 	Input       chunkLength   Output
+// 	[1, 2, 3]   4             [ [1, 2, 3] ]
+// 	[1, 2, 3]   3             [ [1, 2, 3] ]
+// 	[1, 2, 3]   2             [ [1, 2], [3] ]
+// 	[1, 2, 3]   1             [ [1], [2], [3] ]
+// 	[1, 2, 3]   0             [ [] ]
```

Comment:
What is the justification behind a zero chunk size returning a slice of one element containing no elements - even if the original slice was empty?

If the behavior isn't clear for an input it should be disallowed. Which means to panic. Rather than have someone learn the weird behavior, depend on it then we can never change it later.

Reply:
> If the behavior isn't clear for an input it should be disallowed. Which means to panic. Rather than have someone learn the weird behavior, depend on it then we can never change it later.
This is a feasible practice as you say,  also I  think we should consider code consistency. The following code in pie faces the same problem, but return empty slice not use panic.

`func SubSlice[T any](ss []T, start int, end int) (subSlice []T) {
	if start < 0 || end < 0 {
		return
	}

	if start >= end {
		return
	}
       .....
`

Reply:
done, panic: chunkLength should be greater than 0